src/utils/yaml_node_utils.cpp

```diff
@@ -23,6 +23,40 @@
 
 namespace mp = multipass;
 
+namespace
+{
+// remove this utility function once C++20 std::string::ends_with is available
+bool ends_with(const std::string_view search_str, const std::string_view sub_str)
+{
+    if (sub_str.size() > search_str.size())
+    {
+        return false;
+    }
+
+    return std::equal(sub_str.crbegin(), sub_str.crend(), search_str.crbegin());
+}
+
+std::string toggle_instance_id(const std::string& original_instance_id)
+{
+    std::string result_instance_id{original_instance_id};
+    const std::string tweak = "-e";
+
+    // Check if the instance_id already ends with the tweak.
+    if (ends_with(original_instance_id, tweak))
+    {
+        // Tweak found at the string end, remove it.
+        result_instance_id.erase(result_instance_id.size() - tweak.size(), tweak.size());
+    }
+    else
+    {
+        // Tweak not found, append it.
+        result_instance_id += tweak;
+    }
+
+    return result_instance_id;
+}
```

Comment:
I like this, makes everything clearer.

Reply:
The original code simply appending suffix, it might generate lengthy instance-id if the user uses snapshot retore to jump between two states where there is an  extra interfaces difference. This way can avoid the instance id length problem.